ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -67,13 +65,14 @@ public void generate(final Ds3ApiSpec spec, final FileUtils fileUtils, final Pat
         this.fileUtils = fileUtils;
 
         try {
-            final ImmutableList<Enum> allEnums = getAllEnums(spec);
-            final ImmutableSet<String> enumNames = ImmutableSet.copyOf(allEnums.stream().map(Enum::getName).collect(Collectors.toSet()));
-            final ImmutableList<Struct> allStructs = getStructsOrderedList(spec, enumNames);
             final ImmutableList<Request> allRequests = getAllRequests(spec);
+            final ImmutableList<Enum> allEnums = getAllEnums(spec);
+            final ImmutableSet<String> enumNames = EnumHelper.getEnumNamesSet(allEnums);
+            final Queue<Struct> allStructs = new LinkedList<>(getAllStructs(spec, enumNames, allRequests));
```

Comment:
That seems like more of an implementation detail of that algorithm rather than something the top level CodeGenerator needs to know/care about.